sycl/doc/extensions/USM/USM.adoc

```diff
@@ -558,69 +695,56 @@ Return value:: none
 
 '''
 === Unified Shared Memory Information and Descriptors
+==== Pointer Queries
+===== get_pointer_type
+[source,cpp]
+----
+usm::alloc get_pointer_type(const void *ptr, const context &ctxt);
+----
+Parameters::
+  * `const void* ptr` - the pointer to query.
+  * `const sycl::context& ctxt` - the SYCL `context` to which the USM allocation belongs
+Return value:: Returns the USM allocation type for `ptr` if `ptr` falls inside a valid USM allocation.  If `ctxt` is a host `context`, returns `usm::alloc::host`.  Returns `usm::alloc::unknown` if `ptr` is not a valid USM allocation.
 
-
-.Unified Shared Memory Pointer Query
+===== get_pointer_device
 [source,cpp]
 ----
-std::tuple<usm::alloc, sycl::device*> get_pointer_info(const void* ptr);
+sycl::device get_pointer_device(const void *ptr, const context &ctxt);
 ----
 Parameters::
-  * `const void* ptr` - the pointer to query
-Return value:: Returns a `std::tuple` containing the type of the allocation and a pointer to the `device` against which it was allocated.  If this is a host allocation, the `device` pointer will be `nullptr`.
+ * `const void* ptr` - the pointer to query
+ * `const sycl::context& ctxt` - the SYCL `context` to which the USM allocation belongs
+ Return value:: Returns the `device` associated with the USM allocation.  If `ctxt` is a host `context`, returns the host `device` in `ctxt`.  If `ptr` is an allocation of type `usm::alloc::host`, returns the first device in `ctxt`.  Throws an error if `ptr` is not a valid USM allocation.
```

Comment:
What kind of error should I expect here? `sycl::exception`?

Reply:
runtime_error

Reply:
Why not a `sycl::exception`? This is a user facing exception and SYCL functions should throw SYCL exceptions.

Reply:
`sycl::runtime_error` is-a `sycl::exception`

Reply:
Oh right, I thought you meant `std::runtime_error`

Reply:
I failed to fully qualify my type. :)